src/citrine/resources/design_workflow.py

```diff
@@ -42,23 +78,30 @@ def register(self, model: DesignWorkflow) -> DesignWorkflow:
             The newly created design workflow.
 
         """
-        if self.branch_id is None:
-            # There are a number of contexts in which hitting design workflow endpoints without a
-            # branch ID is valid, so only this particular usage is deprecated.
-            msg = ('A design workflow must be created with a branch. Please use'
-                   'branch.design_workflows.register() instead of '
-                   'project.design_workflows.register().')
-            raise RuntimeError(msg)
+        if self.branch_root_id is None or self.branch_version is None:
```

Comment:
Does is ever make sense to have one set and the other not?

Reply:
No. Programmatically, it can happen, but in the context of the collection, we always want them both set.